### DISCUSSION TITLE
Audio Board build issue

### DISCUSSION THREAD

**Discussion by electro991** [CrashOverride85/zc95#114] [Q&A] [answered]
Hello

I have completed the build of 2 audio boards and i must have made a mistake on the build as they are both acting up the same

when i start the unit, every thing is good and it is detected. I can go in audio input config and see a steady line, and a not so steady line if i enable mic pre

the problem i have is as soon as i connect a microphone, the adcs start to act up and move by themselves for all 4 channels and after a couple seconds i get a front panel error “ front panel fault, error reading ADC”

this is one of the assembled boards
![image](https://github.com/user-attachments/assets/ee2afe32-a362-4eb2-8cce-807baaafe421)

and the microphone
![image](https://github.com/user-attachments/assets/67e98dac-46e6-41ee-a9de-52912037a755)

Any help on how i could troubleshoot would be greatly appreciated

**Comment by CrashOverride85** [comment]
I'm not sure, that's not something I've seen before. I'll have a closer look at the photos later, but the only way I can think of that could give you that error when plugging in a mic (and not at power on) is if something causes the 3.3v rail to drop.
Could you check R2 & R24 are 2.2K, and not something much lower?

**Reply by electro991** [reply]
Thank you. I will check and get back to you. There is a chance i may have a problem with a resistor! In some places on the pcb, it was not that easy to make sure which R print was the closest to the hole i soldered in

**Reply by electro991** [reply]
I have reviewed all resistors on the board. Tested all using multimeter, obviously, since soldered, some did not test accurately so I confirmed color code for the 5-6 that didn’t test right value. They are all ok based on this. I have attached the confirmed values, which may also help future builds. You can take a look and comment if you see something wrong.

I am a bit puzzled now… I am wondering about Q1, Q3 and Q2 orientation as well as D3.

Also should I hear the little relay K1 click at any point, is it audible ?

[AudioOutput.pdf](https://github.com/user-attachments/files/16563550/AudioOutput.pdf)

**Reply by CrashOverride85** [reply]
Yes, those resistor values all look correct to me. 
Having re-checked the datasheet, yes, Q1 & Q3 look to be the correct orientation (i.e. backwards to the silkscreen).
Mostly unrelated, but I've realised the notes about this in the build notes aren't right - the silkscreen is just backwards; I'm not sure why I thought bs250 vs bs250p made a difference. I can see you've got the bs250p so no issue.

Regardless, even if they were backwards, all it would mean is that mic power would be permanently on, which with a mic plugged would be ok.

Q2 looks good too.

The relay is fairly quiet, but you should be able to hear the relay click at power on, and when you go into "Config -> Hardware Config -> Aux port use" and change between Audio and Serial - it should click in realtime as you change the option.

If you've got a spare 3.5mm connector, it might be worth seeing what happens if you short Tip-> Sleeve and then Ring->Sleeve, and see if one or both of those combinations cause the same issue.

**Comment by electro991** [comment]
Hello again,

As a first step, I have been trying to try to hear that relay click and I can't hear anything as I move the Aux Port use. So I am thinking I either have an the incorrect relay or whatever triggers this relay to click is not operating. If the relay is not switching, I am connecting a mic to the ttl to serial chip? Maybe that is what is causing the issue while everything else appears working (board detected etc)

I tried 5v on a bare G6K-2F-Y-TR and I definately hear the click so I know I would hear its sound when playing with the setting. Do you have any suggestion to further troubleshoot this relay ? Thanks a lot for your help.

**Reply by electro991** [reply]
I inspected the schematic. Possibly the 2n7000 is incorrect part or faultly. With multimeter, I have continuity between J1 5v and relay pin 1. Contuinity from AudioEN on J1 to middle pin of 2n7000. Looks like the 2n7000 is not latching the relay.

**Reply by CrashOverride85** [reply]
yes, with the relay not powered it'll be connecting to mic to the 232 IC, that shouldn't cause the issue you're seeing. That IC should be pretty tolerant of all sorts tbh. It certainly is starting to sound like something's up with that 2n7000, but I still can't think how this could be causing the issue you're seeing with the ADC issues.

**Reply by electro991** [reply]
Hello again. Would you be able to send a picture of the orientation of D3 please ?

**Reply by electro991** [reply, marked as answer]
Issue Resolved. I had D3 reversed which was causing a short circuit when Q2 was fully on. That was the reason the relay was not being latched and possible my issue described above.

I believe nothing else got damaged so I replace Q2 and D3 for new components.

To orient D3, I was looking at the silkscreen "A" and put the anode on that side of the board, but based on schematic, its a flyback and should be the other way. Is this accurate ?

**Reply by CrashOverride85** [reply]
Oh wow, good spot!
It's the wrong way round on my board too which is why I didn't spot it in your photo. But having tested the diode (only in circuit so far), my diode seems to be dead, which I assume is why it works

**Reply by electro991** [reply]
Your 2n7000 might have overheated too. I did turn it fully on with 5v and with that diode reversed, the 2n7000 became smoking hot. I changed it just in case.

**Reply by CrashOverride85** [reply]
Yeah, think I'm going to replace both the diode and 2n7000 this weekend, although it somehow appears to have survived (relay switches when it's supposed to)

**Reply by electro991** [reply]
I did some more testing and it appears to me that the 2n7000s i have are not getting enough voltage on 3.3 volt to become fully on thus causing the relay to arbitrarily click on and off when applying some power on channels (this even happening while not using audio patterns). I hear the relay constantly clicking on off. It appears that 3.3 volt is too borderline for the 2n7000. Reading about this on google, it seems common on 3.3 volt for this part. 

If i am understanding what happening correctly, I think an ao3400 would be more of an appropriate part for the purpose but the are smd. 

I will try to get a little adapter pcb made to try it on my existing boards. 

Let me know if you encounter this at any point.

**Reply by electro991** [reply]
I made this little board that will let me try an AO3400. Its in production should get it in a week or 2

![image](https://github.com/user-attachments/assets/24a3e473-fbc2-4157-8184-a7fe3c82fb3a)

**Reply by CrashOverride85** [reply]
You're right, whilst I've not had the same issue as you with the 2n7000, I agree it's a very poor choice here :(

I'm sightly surprised it's arbitrarily clicking on and off though, if there's enough current to activate it, I'd have thought it'd stay on due to the holding current requirement being less.

The AO3400 seems like a good choice, and it's the same nfet already being used on the output board, controlled from 3.3v and driving a higher load than the relay.

I'm struggling to find any suitable replacement in a TO-92 package :/ maybe ZVN2106A. Relay only needs ~20ma, should be more suitable than 2n7000, but still not great.

I think the answer is going to be to redesign the board, either as SMD and use the AO3400, or stick with through hole and use a BJT.

**Reply by electro991** [reply]
Thank you. I will report back once i installed the ao3400 and see if everything works better after

**Reply by electro991** [reply]
Looping back on my findings… it seems that there may be a number of issues driving that relay. 

I have assembled 3 audio boards suspecting it could be my build but it appears that all 3 are having issues driving the relay consistently. I even had at times a flickering relay on and off constantly. Tried to replace the pcf8574 on the main board etc and issue persisting. I am not sure if there are many people that actually built the audio board…

In summary, here’s what i found;

1. the 2n7000 and ao3400 are both having issues acting as a switch
2. i monitored the voltage on 3.3 as well as on AUDIO_EN and I noticed that as soon as you start a pattern, the 3.3 volt line oscillates between 3.29 to 3.1 and even hit 2.8 at times. Might be the load and limitation of the small 3.3 volt regulator. When this happens, the fet seems to struggle. 
3. Reading about the pcf8574, voltage can be weak when used as output, I think this is contributing to the issue keeping the fet “On”

This behaviour happen on my 2 zc95, trying the different combinations of audio board i build with 2n7000 and AO3400 driving the relay coil. 

So… i gave up and removed the fet, adding a jumper between pin 1 and 3! Relay is closed permanently and works properly

i am not sure what other solution could work as i strongly believe there a combination of problems starting with the 3.3 volt regulator down to the pcf8574… maybe upgrading the regulator could fix this, not sure

hope this helps

**Reply by CrashOverride85** [reply]
This is certainly starting to get quite confusing, I'm not sure what's going on. I'll have a proper look at schematic later and see if I can spot the potential issue.

But in the meantime....
* The AO3400 should be _more_ than capable of switching the relay
* The 3.3 rail dropping when starting a pattern is super weird, as it's not used by the output board (that has its own 3.3v regulator)
* The 3.3v regulator is good for ~100mA, which really should be plenty
* Whilst the pcf8574 can't supply much current, switching a mosfet like a AO3400 or 2n7000 requires basically nothing

I think the most interesting thing is that the 3.3v rail is dropping when starting pattern, that really should not be happening. Could you measure the 5v supply? Is that also dropping? Also, is the 3.3 regulator getting warm/hot?

**Reply by electro991** [reply]
Thanks for the idea, i redid my measurements and it turns out 3.3 was stable on the rail but unstable on the PCF output. I changed the PCF and it is now stable on the output. A03400 now stays on !

**Comment by electro991** [comment]
Problem has come back randomly, the pcf is unable to maintain the ao3400 « on » causing the relay to flicker on and off somewhat rapidly at times. Sometimes it holds for 10-15 minutes then it starts to flicker on and off

Again, i am far from an expert in this but I have done a lot of research and reading the pcf8574 datasheet and believe this is where the issue is. Because the pcf has a quasi-bidirectional output, it doesn’t have the same capabilities as a typical push-pull pin of an mcu. This becomes an issue here because the pin is used as output to drive the fet. The ao3400 doesn’t need much to switch on, however i think its only working under the best conditions and there isn’t much tolerance for it to work. 

The fet needs voltage to turn on but the guaranteed min voltage on output for the pcf is less than vcc, like -0.5 vcc.  

Also, for current the minimum guaranteed on a pcf pin is only 30 µA. Just R27, the pull down,  drives 33 µA at 3.3v. Maybe the resistor is overloading the output by itself.  

That said, not sure if any of this makes sense but reading the datasheet, their output examples don’t have a pulldown which I believe in this case is not helping the problem. So i removed R27 from my audio board. So far, this change seems to have helped. Time will tell as the issue was a bit random ! It just hasn’t occurred since.

**Reply by CrashOverride85** [reply]
You're right, I don't think I'd realised quite how little current the pcf8574 could source. 
With it defaulting to high on power on, I guess there's no need for a resistor at all, but it should have been a pull-up not a pull-down if at all.
I still think the AO3400 is a good choice here, and the limited current from the pcf8574 shouldn't be an issue - it might take _slightly_ longer to switch on than ideally it would, but not long enough to cause a problem given it's not high frequency or switching much current. And once on, it should be taking nothing to keep it on. It's that resistor as you calculated is likely the issue.

**Reply by electro991** [reply]
I can confirm that the removal of R27 along with the AO3400 has completely resolved the issue!

**Reply by CrashOverride85** [reply]
Excellent, glad to hear it and thanks for letting me know :)
I'll update the build notes shortly to not include that resistor, and think about redesigning it
